### ISSUE TITLE
Remove usage of twisted.python.compat.set and frozenset

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6297]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6297|
|Type|task|
|Created|2013-02-11 22:44:18Z|
|Branch|https://github.com/twisted/twisted/tree/remove-set-frozenset-6297|

`set` and `frozenset` builtins were introduced in python2.4 and the `sets` module deprecated since python2.6. This makes `compat.set` and `compat.frozenset` redundant and the following imports can be removed:

```
./twisted/trial/runner.py:26:from twisted.python.compat import set
./twisted/trial/reporter.py:21:from twisted.python.compat import set
./twisted/trial/test/test_script.py:13:from twisted.python.compat import set
./twisted/scripts/trial.py:15:from twisted.python.compat import set
./twisted/mail/relaymanager.py:33:from twisted.python.compat import set
./twisted/mail/maildir.py:21:from twisted.python.compat import set
./twisted/web/_newclient.py:36:from twisted.python.compat import set
./twisted/web/test/test_wsgi.py:17:from twisted.python.compat import set
./twisted/web/test/test_static.py:13:from twisted.python.compat import set
./twisted/web/client.py:39:from twisted.python.compat import set
./twisted/protocols/amp.py:185:from twisted.python.compat import set
./twisted/names/test/test_client.py:12:from twisted.python.compat import set
./twisted/python/test/test_zipstream.py:12:from twisted.python.compat import set
./twisted/python/test/test_fakepwd.py:23:from twisted.python.compat import set
./twisted/internet/posixbase.py:18:from twisted.python.compat import set, _PY3
./twisted/internet/base.py:20:from twisted.python.compat import set
./twisted/internet/_glibbase.py:23:from twisted.python.compat import set
./twisted/internet/test/test_posixbase.py:10:from twisted.python.compat import set, _PY3
./twisted/internet/test/test_process.py:14:from twisted.python.compat import set
./twisted/internet/iocpreactor/reactor.py:35:from twisted.python.compat import set
./twisted/conch/test/test_openssh_compat.py:12:from twisted.python.compat import set
./twisted/conch/test/test_tap.py:27:from twisted.python.compat import set
./twisted/test/test_compat.py:15:from twisted.python.compat import set, frozenset, reduce, execfile, _PY3
./twisted/test/test_process.py:31:from twisted.python.compat import set
./twisted/test/test_paths.py:14:from twisted.python.compat import set, _PY3
./twisted/test/test_task.py:10:from twisted.python.compat import set
./twisted/words/protocols/irc.py:44:from twisted.python.compat import set
```

The `compat` versions could also be deprecated if deemed necessary.

Attachments:

* [kevinwo-6297.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f7a/f7aae5c842f70db90e57cf702215d888c23228e9/4cb51f86e66426049769a48fd3cfeeb7c054e6aa.patch) (13142 bytes) - added by kevinwo on 2013-02-27 02:25:22Z - 
* [remove-set-frozenset-kevinwo-6297-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f7a/f7aae5c842f70db90e57cf702215d888c23228e9/0de126fc3722fa692c22f9db9f6635ab70ddcf8c.patch) (17699 bytes) - added by kevinwo on 2013-03-06 01:05:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6297 6297
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_remove_set_frozenset_6297 branches/remove-set-frozenset-6297
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1360622658000000 1360622658000000
changetime__1362540211000000 1362540211000000
version__None None
owner__tom_prince tom.prince
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="kevinwo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|kevinwo set owner to kevinwo<br>kevinwo set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="kevinwo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|kevinwo removed owner<br>kevinwo set status to `new`|
|-|-|

All traces of 'set' and 'frozenset' have been removed

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @tomprince|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince set owner to kevinwo|
|-|-|

Thanks for your contribution.

There are a couple of more uses:

    twisted/web/test/test_http_headers.py
    twisted/web/test/test_http.py
    twisted/python/test/test_release.py 

Also, you need to add  news files (.misc probably) to each touched component.

Please resubmit for review after removing the remaining uses and adding a topfile.

**Comment by twisted-trac** [comment]
|<img alt="kevinwo's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|kevinwo set owner to @tomprince|
|-|-|

Thanks, Tom. I've attached an updated patch handling the uses you mentioned along with topfiles.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince commented|
|-|-|

(In [37414]) Branching to remove-set-frozenset-6297.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_9"></a>|@tomprince commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-set-frozenset-6297)

Just a note: .misc topfiles don't need (and ignore) conetent. And for the rest of the topfiles, they should contain something appropriate to go in the NEWS files and they get rewrapped. For this particular change, since there is no user visibile change, the misc file just causes the ticket to appear under "Other".

I'll merge.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_10"></a>|@tomprince set status to `closed`|
|-|-|

(In [37416]) Merge remove-set-frozenset-6297: Remove usage of twisted.python.compat.set and frozenset.

Author: kevinwo
Reviewers: therve
Fixes: [#6297](https://github.com/twisted/twisted/issues/6297)

set and frozenset builtins were introduced in python2.4 and
the sets module deprecated since python2.6. This makes
compat.set and compat.frozenset redundant.
